lib/convection/model/template/resource/aws_iam_role.rb

```diff
@@ -32,14 +32,24 @@ def with_instance_profile(&block)
             @template.resources[profile.name] = profile
           end
 
-          ## Add a canned trust policy for EC2 instnces
+          ## Add a canned trust policy for EC2 instances
           def trust_ec2_instances(&block)
             @trust_relationship = Model::Mixin::Policy.new(:name => 'trust-ec2-instances', :template => @template)
             trust_relationship.allow do
               action 'sts:AssumeRole'
               principal :Service => 'ec2.amazonaws.com'
             end
-
+            trust_relationship.instance_exec(&block) if block
+            trust_relationship
+          end
+	
+	## Add Add a canned trust policy for Cloudtrail
+	def trust_cloudtrail(&block)
+            @trust_relationship = Model::Mixin::Policy.new(:name => 'trust-cloudtrail-instances', :template => @template)
+            trust_relationship.allow do
+              action 'sts:AssumeRole'
+              principal :Service => 'cloudtrail.amazonaws.com'
+            end
```

Comment:
Where would this be applied?  It might be worth adding a test or example to illustrate it's usage.